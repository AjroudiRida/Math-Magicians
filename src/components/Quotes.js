import { useEffect, useState } from 'react';
import './quotes.css';

const Quotes = () => {
  const [quotes, setQuotes] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getQuotes = async () => {
      await fetch(
        'https://api.api-ninjas.com/v1/quotes?category=success&&limit=10',
        {
          headers: { 'X-Api-Key': '4jSXs4kYBN2NYbxRcEH+pQ==9pROL1fNIxazrI0m' },
          contentType: 'application/json',
        },
      )
        .then((res) => res.json())
        .then((jsonData) => {
          setQuotes(jsonData);
          setIsLoading(false);
        })
        .catch((err) => err);
    };

    setTimeout(() => {
      getQuotes();
    }, 2000);
  }, []);

  return (
    <>
      <h1 className="quotes-title">get inspired by this success quotes</h1>
      {(isLoading === false) ? (
        quotes.map((item) => (
          <blockquote key={item.quote.toString()} className="quote-text">
            <p className="quote">{item.quote}</p>
            <footer>
              <cite>
                -
                {item.author}
              </cite>
            </footer>
          </blockquote>
        ))
      ) : (
        <div>
          <blockquote className="blockquote-loading">
            <p className="quote-loading skeleton" />
            <footer>
              <cite className="author-loading skeleton" />
            </footer>
          </blockquote>

          <blockquote className="blockquote-loading">
            <p className="quote-loading skeleton" />
            <footer>
              <cite className="author-loading skeleton" />
            </footer>
          </blockquote>
        </div>
      )}
    </>
  );
};

export default Quotes;
