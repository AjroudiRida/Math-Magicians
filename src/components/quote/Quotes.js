import { useEffect, useState } from 'react';
import './quotes.css';

const Quotes = () => {
  const [quotes, setQuotes] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getQuotes = async () => {
      await fetch(
        'https://api.api-ninjas.com/v1/quotes?category=success&&limit=10',
        {
          headers: { 'X-Api-Key': '4jSXs4kYBN2NYbxRcEH+pQ==9pROL1fNIxazrI0m' },
          contentType: 'application/json',
        },
      )
        .then((res) => res.json())
        .then((jsonData) => {
          setQuotes(jsonData);
          setIsLoading(false);
        })
        .catch((err) => err);
    };
    getQuotes();
  }, []);

  return (
    <div className="main">
      <h1 className="quotes-title">get inspired by this success quotes</h1>
      {isLoading === false ? (
        quotes.map((item) => (
          <blockquote
            data-testid="qte"
            key={item.quote.toString()}
            className="quote-text"
          >
            <p data-testid="qt" className="quote">
              {item.quote}
            </p>
            <footer>
              <cite>
                -
                {item.author}
              </cite>
            </footer>
          </blockquote>
        ))
      ) : (
        <div>
          <blockquote className="blockquote-loading">
            <p className="quote-loading skeleton" />
            <footer>
              <cite className="author-loading skeleton" />
            </footer>
          </blockquote>

          <blockquote className="blockquote-loading">
            <p className="quote-loading skeleton" />
            <footer>
              <cite className="author-loading skeleton" />
            </footer>
          </blockquote>
        </div>
      )}
    </div>
  );
};

export default Quotes;
