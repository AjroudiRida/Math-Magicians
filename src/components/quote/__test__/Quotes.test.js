import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Quotes from '../Quotes';
const quotes = [
  { quote: 'quote 1', author: 'author 1' },
  { quote: 'quote 2', author: 'author 2' },
];
beforeEach(() => {
  jest.spyOn(window, 'fetch');
});
it('should renders an h1 heading in the document', () => {
  render(<Quotes />);
  const titleElement = screen.getByRole('heading', {
    name: 'get inspired by this success quotes',
  });
  expect(titleElement).toBeInTheDocument();
});
it('should render 1 quotes', async () => {
  const mockResponse = { json: async () => quotes };
  window.fetch.mockResolvedValue(mockResponse);
  render(<Quotes />);
  await waitFor(() => {
    const quoteElements = screen.getAllByTestId('qte');
    expect(quoteElements).toHaveLength(2);
  });
});
it('should match the Dom snapshot', () => {
  const quoteTree = renderer.create(<Quotes />).toJSON();
  expect(quoteTree).toMatchSnapshot();
});
