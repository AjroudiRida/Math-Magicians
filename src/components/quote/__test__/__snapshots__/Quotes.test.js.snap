// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the Dom snapshot 1`] = `
<div
  className="main"
>
  <h1
    className="quotes-title"
  >
    get inspired by this success quotes
  </h1>
  <div>
    <blockquote
      className="blockquote-loading"
    >
      <p
        className="quote-loading skeleton"
      />
      <footer>
        <cite
          className="author-loading skeleton"
        />
      </footer>
    </blockquote>
    <blockquote
      className="blockquote-loading"
    >
      <p
        className="quote-loading skeleton"
      />
      <footer>
        <cite
          className="author-loading skeleton"
        />
      </footer>
    </blockquote>
  </div>
</div>
`;
